docs/pages/world/modules/keysintable.mdx

```diff
@@ -0,0 +1,239 @@
+
+For example, here is a Solidity script that reads the tasks added to `Tasks` after module installation.
+
+<CollapseCode>
+
+```solidity filename="UseGetKeysInTable.s.sol" copy showLineNumbers {9,19-23,25,27-33,35-39}
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.21;
+
+import { Script } from "forge-std/Script.sol";
+import { console } from "forge-std/console.sol";
+import { StoreSwitch } from "@latticexyz/store/src/StoreSwitch.sol";
+
+import { IWorld } from "../src/codegen/world/IWorld.sol";
+import { getKeysInTable } from "@latticexyz/world-modules/src/modules/keysintable/getKeysInTable.sol";
+import { Tasks, TasksData } from "../src/codegen/index.sol";
+
+contract UseGetKeysInTable is Script {
+  function run() external {
+    uint256 deployerPrivateKey = vm.envUint("PRIVATE_KEY");
+    address worldAddress = vm.envAddress("WORLD_ADDRESS");
+    IWorld world = IWorld(worldAddress);
+    StoreSwitch.setStoreAddress(worldAddress);
+
+    vm.startBroadcast(deployerPrivateKey);
+    world.app__addTask("Walk the cat");
+    world.app__addTask("Cook delicious meal");
+    world.app__addTask("Practice Elvish");
+    vm.stopBroadcast();
+
+    bytes32[][] memory keys = getKeysInTable(Tasks._tableId);
+
+    console.log("Number of keys:", keys.length);
+    for (uint i = 0; i < keys.length; i++) {
+      console.log("\nKey #", i, ":");
+      for (uint j = 0; j < keys[i].length; j++) {
+        console.logBytes32(keys[i][j]);
+      }
+    }
+
+    console.log("\n\nTask descriptions");
+    for (uint i = 0; i < keys.length; i++) {
+      TasksData memory task = Tasks.get(keys[i][0]);
+      console.log(task.description);
+    }
+  }
+}
+```
+
+</CollapseCode>
+
+<details>
+
+<summary>Explanation</summary>
+
+```solidity
+import { getKeysInTable } from "@latticexyz/world-modules/src/modules/keysintable/getKeysInTable.sol";
+```
+
+The function that gets the keys in the table.
+
+```solidity
+    StoreSwitch.setStoreAddress(worldAddress);
+```
+
+To use `getKeysInTable` we need to set the store address to `worldAddress`.
+
+```solidity
+    vm.startBroadcast(deployerPrivateKey);
+    world.app__addTask("Walk the cat");
+    world.app__addTask("Cook delicious meal");
+    world.app__addTask("Practice Elvish");
+    vm.stopBroadcast();
+```
+
+Creates three entries with different descriptions, to ensure we have entries to show.
+Note that this is the _only_ place in the script we need to send transactions.
+`getKeysInTable` is [a `view` function](https://docs.soliditylang.org/en/v0.8.26/contracts.html#view-functions), and does not change the state.
+
+```solidity
+    bytes32[][] memory keys = getKeysInTable(Tasks._tableId);
+```
+
+Get the key(s) by passing the Tasks table id as a parameter.
+
+```solidity
+    console.log("Number of keys:", keys.length);
+    for(uint i=0; i<keys.length; i++) {
+       console.log("\nKey #", i, ":");
```

Comment:
```suggestion
       console.log("Key #", i, ":");
```
I think `console.log` already adds a new line?
(need to update the above script to match, too)

Reply:
`console.log` already adds *a* new line. But I want here to distinguish between different table rows more than I distinguish between different key fields of the same entry.

```
    for(uint i=0; i<keys.length; i++) {
       console.log("\nKey #", i, ":");
       for(uint j=0; j<keys[i].length; j++) {
          console.logBytes32(keys[i][j]);
       }
    }
```

Gives us 

```
Key #0:
0xdead
0xbeef

Key#1:
0x0bad
0x60a7
```

I think that is better than 

```
Key #0:
0xdead
0xbeef
Key#1:
0x0bad
0x60a7
```